### DISCUSSION TITLE
Unable to plot EBSD data

### DISCUSSION THREAD

**Discussion by sarialzerey** [mtex-toolbox/mtex#2036] [Ask Anything]
Hi all,

I'm performing basic plot command:
plot(ebsd('copper'),ebsd('copper').orientations,'micronbar','on')

but I receive a message saying:
error using  ./ 
Arrays have incompatible sizes for this operation.
![Capture2](https://github.com/mtex-toolbox/mtex/assets/141929646/a4f4d76f-1f7d-498e-80c8-b73c244d5cf0)
![Capture1](https://github.com/mtex-toolbox/mtex/assets/141929646/c7e870e9-4a5d-484b-b498-74366e72d882)

**Comment by zmichels** [comment]
Hmmm... not sure without seeing the data... but I would try a couple things:

Does an even simpler command like:
plot(ebsd)
work correctly with your dataset?

What about:
gridify(ebsd)
?

If that works... does the following improve anything for plotting?
e = gridify(ebsd)
plot(e('copper'), e('copper').orientations)

Also,have you tried other datasets to confirm the issue is with the data? Like do you receive a similar error after loading an example dataset and trying something similar?
mtexdata forsterite
plot(ebsd('f'),ebsd('f').orientations)

**Reply by sarialzerey** [reply]
Thank you for the swift reply. I'm really in trouble because I have a due date by Monday.
I have 50 files (80GB)  that I scanned in a single session that none of them work. 

I have previous measurements that have no problem. 
e = gridify(ebsd)
plot(e('copper'), e('copper').orientations)  does not help.

There was a message from Bruker's scanning software saying that there won't be a scale bar in the measurement because of configuration issue between the SEM and the EBSD scanning software (Espirit 2.5).
One thing I noticed that the problematic files don't have unitcell information but the good measurement has. When I did override the unitcell, the plot command generated an image of a single solid color.
I attached both the EBSD arrays in the screenshot. The good one is on the right whereas the problematic is on the left
![left is bad right is good](https://github.com/mtex-toolbox/mtex/assets/141929646/88223c9f-cf30-43c2-886c-3dc03eee107d)

**Reply by kilir** [reply]
Maybe you can share one of those problemtic files.
Cheers,
Rüdiger

**Reply by sarialzerey** [reply]
Sure!

https://drive.google.com/file/d/1pp-Ix3isiEnvVtGi3Sj9SWSkilzbbRvj/view?usp=drivesdk

**Comment by kilir** [comment]
Can you test this commit and see if it does not break other files which used to work?
581857ce62001dd5acad0575ea034def4f0a2b0b
Cheers,
Rüdiger

**Reply by sarialzerey** [reply]
I'm unable to run this code
![image](https://github.com/mtex-toolbox/mtex/assets/141929646/c51d6ac2-455b-4ab4-b033-2699f6e62755)

**Reply by kilir** [reply]
You need to replace the function 'loadEBSD_h5' in your mtex installation with the entire function from the link. What you try is running the changes which doesnot work.

https://github.com/mtex-toolbox/mtex/blob/develop/interfaces%2FloadEBSD_h5.m
